src/modules/restaurants/components/SpecificRestaurantPage.tsx

```diff
@@ -54,28 +54,21 @@ function SpecificRestaurantPage() {
 	};
 
 	useEffect(() => {
-		async function getRestaurantByID() {
+		async function getRestaurantByName() {
 			try {
-				const result = await getRestaurantByIDFromAPI(restaurantName);
+				const result = await getRestaurantByNameFromAPI(restaurantName);
 				setRestaurant(result.data);
 			} catch (error: unknown) {
 				console.log(error);
 			}
 		}
-		getRestaurantByID();
+		getRestaurantByName();
 	}, []);
 
 	useEffect(() => {
-		async function getAllDishesFromCategory() {
-			try {
-				const result = await getAllDishesFromCategoryFromAPI(selectedMealCategory);
-				setAllDishesFromCategory(result.data);
-			} catch (error: unknown) {
-				console.log(error);
-			}
-		}
-		getAllDishesFromCategory();
-	}, [selectedMealCategory]);
+		const filteredDishes = restaurant?.dishes.filter((dish) => dish.mealType.includes(selectedMealCategory));
```

Comment:
We need to think about where we should put the filters?
https://stackoverflow.com/questions/8368002/search-filter-should-i-do-it-client-side-or-server-side